TWTValidation/JSON Validator/Parser/TWTJSONRemoteSchemaManager.m

```diff
@@ -0,0 +1,95 @@
+//
+//  TWTJSONRemoteSchemaManager.m
+//  TWTValidation
+//
+//  Created by Jill Cohen on 10/19/15.
+//  Copyright © 2015 Ticketmaster Entertainment, Inc. All rights reserved.
+//
+
+#import "TWTJSONRemoteSchemaManager.h"
+
+#import <TWTValidation/TWTJSONSchemaASTCommon.h>
+#import <TWTValidation/TWTJSONSchemaParser.h>
+
+
+@interface TWTJSONRemoteSchemaManager ()
+
+@property (nonatomic, strong) NSMutableDictionary *filePathsToJSONSchemaTopLevelNodes;
+
+@end
+
+
+@implementation TWTJSONRemoteSchemaManager
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        _filePathsToJSONSchemaTopLevelNodes = [[NSMutableDictionary alloc] init];
+    }
+    return self;
+}
+
+
+- (BOOL)attemptToConfigureFilePath:(NSString *)filePath onReferenceNode:(TWTJSONSchemaReferenceASTNode *)referenceNode
```

Comment:
This method is a little weird to me. It makes me nervous that we’re modifying the reference node’s file path, especially since we seem to do it even if there’s a failure.

Let’s change it so that it doesn’t set the path unless we can fetch it. We might also surface the `NSError *` that the fetch method swallows right now and rename the method something like:

```objc
- (BOOL)validateAndSetFilePath:(NSString *)filePath onReferenceNode:(TWTJSONSchemaReferenceASTNode *)referenceNode error:(NSError **)errorOut;
```